tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -0,0 +1,77 @@
+// EnergyPlus::WaterThermalTank Unit Tests
+#include <cmath>
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include <WaterThermalTanks.hh>
+
+using namespace EnergyPlus;
+
+bool FloatEqualTest(Real64 val1, Real64 val2, Real64 SmallNum = 0.00000001) {
+	using namespace std;
+	return ( abs(val1 - val2) < SmallNum );
+}
+
+TEST( HeatPumpWaterHeaterTests, TestQsourceCalcs )
+{
+	Real64 DeltaT = 0.0;
+	Real64 const SourceInletTemp = 62.0;
+	Real64 const Cp = 4178.; // water, J/(kg * K)
+	Real64 const SetPointTemp = 60.0;
+	Real64 const SourceMassFlowRateOrig = 0.378529822165; // water, 6 gal/min
+	Real64 SourceMassFlowRate = SourceMassFlowRateOrig;
+	Real64 Qheatpump = 0.0;
+	Real64 Qsource = 0.0;
+	
+	// Mixed Tank
+	
+	// Test case without HPWH
+	WaterThermalTanks::CalcMixedTankSourceSideHeatTransferRate(DeltaT, SourceInletTemp, Cp, SetPointTemp, SourceMassFlowRate, Qheatpump, Qsource);
+	// Qsource is non zero and calculated relative to the tank setpoint.
+	EXPECT_TRUE(FloatEqualTest(SourceMassFlowRate * Cp * (SourceInletTemp - SetPointTemp), Qsource));
```

Comment:
@nmerket Instead of writing your own floating point comparison function, i'd suggest using EXPECT_DOUBLE_EQ() [1] instead of EXPECT_TRUE(). Google Test already has floating point comparison built into it for assertions. This avoids the need for everyone to write a (or use your) floating point comparison function, especially since floating point comparisons are surprisingly difficult to get right [2].

[1] https://code.google.com/p/googletest/wiki/V1_7_AdvancedGuide#Floating-Point_Comparison
[2] http://www.cygnus-software.com/papers/comparingfloats/comparingfloats.htm